wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/SnakImpl.java

```diff
@@ -66,10 +70,37 @@
 	 */
 	private final PropertyIdValue property;
 	
+	/**
+	 * The wikidata hash of this Snak . null if we don't have knowledge about the hash.
+	 */
+	private String hash;
+	
+	
+	/**
+	 * The snaktype of this snak.
+	 */
+	private final String snaktype;
+	
+	/**
+	 * The datavalue of this snak.
+	 */
+	private final HashMap<String, String> datavalue;
+	
+	/**
+	 * The datatype of this snak.
+	 */
+	private final String datatype;
```

Comment:
Same here.

Reply:
Okay, thank you for your comments, I will make the necessary changes and commit again, Also, any comments for the testing part?